lib/ansible/modules/network/nxos/nxos_vsan.py

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+
+DOCUMENTATION = '''
+module: nxos_vsan
+extends_documentation_fragment: nxos
+version_added: 2.9
+short_description: Configuration of vsan.
+description:
+    - Configuration of vsan for Cisco MDS NXOS.
+author:
+    - Suhas Bharadwaj (@srbharadwaj) (subharad@cisco.com)
+options:
+    vsan:
+        description:
+            - List of vsan details to be added or removed
+        type: list
+        suboptions:
+            id:
+                description:
+                    - Vsan id
+                required: True
+            name:
+                description:
+                    - Name of the vsan
+            suspend:
+                description:
+                    - suspend the vsan if True
+                type: bool
+                default: False
+            remove:
+                description:
+                    - Removes the vsan if True
+                type: bool
+                default: False
+            interface:
+                description:
+                    - List of vsan's interfaces to be added
+                type: list
+'''
+
+EXAMPLES = '''
+---
+-
+  name: "Test that vsan module works"
+  nxos_vsan:
+    provider: "{{ creds }}"
+    vsan:
+      -
+        id: 922
+        interface:
+          - fc1/1
+          - fc1/2
+          - "port-channel 1"
+        name: vsan-SAN-A
+        remove: false
+        suspend: false
+      -
+        id: 923
+        interface:
+          - fc1/11
+          - fc1/21
+          - "port-channel 2"
+        name: vsan-SAN-B
+        remove: false
+        suspend: true
+      -
+        id: 1923
+        name: vsan-SAN-Old
+        remove: true
+
+'''
+
+RETURN = '''
+commands:
+  description: commands sent to the device
+  returned: always
+  type: list
+  sample:
+    - terminal dont-ask
+    - vsan database
+    - vsan 922 interface fc1/40
+    - vsan 922 interface port-channel 155
+    - no terminal dont-ask
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.nxos.nxos import load_config, nxos_argument_spec, run_commands
```

Comment:
We are not to support connection local for new modules, please remove `nxos_argument_spec` import.